sqs/provider.go

```diff
@@ -192,6 +193,48 @@ func (s *Provider) Bind(ctx context.Context, bindData provideriface.BindData) (*
 		return nil, err
 	}
 
+	if !bindData.AsyncAllowed {
```

Comment:
That's interesting: what were we previously doing if the request was `!bindData.AsyncAllowed`? Just saying screw you and doing asynchronously anyway?

Reply:
Yes we were just ignoring it.

If this poll/wait thing doesn't work ... we will add logic to say `computer says no`